### PR TITLE
general/bool: Remove TRUE/FALSE and boolean-style uses with int types

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #9298

Describe changes:
- Remove TRUE/FALSE and replace with logic/code using bool native types

Updates:
- Rebase

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16279

File: src/runmodes.c [hunk 1]
```diff
@@ -982,12 +982,15 @@ float threading_detect_ratio = 1;
...
 void RunModeInitializeThreadSettings(void)
 {
-    threading_set_cpu_affinity = FALSE;
-    if ((ConfGetBool("threading.set-cpu-affinity", &threading_set_cpu_affinity)) == 0) {
-        threading_set_cpu_affinity = FALSE;
+    int affinity = 0;
+    if ((ConfGetBool("threading.set-cpu-affinity", &affinity)) == 0) {
+        threading_set_cpu_affinity = false;
+    } else {
+        threading_set_cpu_affinity = affinity == 1;
```
Comment (inashivb) [inline src/runmodes.c:989]: Is this a better way of doing the same thing in C? Trying to justify the addition of new var here.

**Review by inashivb — COMMENTED** [review]
Looks good to me. :)
Just a question inline.

File: src/runmodes.c (hunk 1 above)
Comment (jlucovsky) [reply src/runmodes.c:989]: `ConfGetBool` doesn't return a C `bool` -- thus `affinity` is an integer type that is then used to set the global (to the source module) scoped `bool threading_set_cpu_affinity`

**Review by jasonish — APPROVED** [review]
Nice cleanup.

**Review by victorjulien — APPROVED** [review]
Approved for 8

**Review by victorjulien — CHANGES_REQUESTED** [review]
Needs rebase

**Comment by jlucovsky** [comment]
Continued in #9804

**Pull request closed without merging**